### DISCUSSION TITLE
Decimal is getting in the way

### DISCUSSION THREAD

**Discussion by omar-dulaimi** [MichalLytek/typegraphql-prisma#103] [Q&A]
Hello,

we're facing issues regarding our upgrade from Prisma 1 to Prisma 2. Since Prisma 1 allowed sending decimal values as numbers, but not anymore with Prisma 2 after they have introduced the DecimalJS dependency. We're trying to maintain the old behavior of sending numbers, since changing it on every frontend project we have would take so much time and might actually introduce new bugs to the systems. 

One of the comments on Prisma issues suggested this solution that they currently use in their company:
```
import { Decimal } from '@prisma/client/runtime'

// recursive function looping deeply throug an object to find Decimals
const transformDecimalsToNumbers = (obj: any) => {
  if (!obj) {
    return
  }

  for (const key of Object.keys(obj)) {
    if (Decimal.isDecimal(obj[key])) {
      obj[key] = obj[key].toNumber()
    } else if (typeof obj[key] === 'object') {
      transformDecimalsToNumbers(obj[key])
    }
  }
}

export const decimalCastMiddleWare = async (resolve, root, args, context, info) => {
  const result = await resolve(root, args, context, info)
  transformDecimalsToNumbers(result)
  return result
}
```

So we want to give it a try. But as you know it's not going to work without an aweful lot of changes to the generated resolevers as well, since Decimal is used in a lot of places, we have to manually change the returned type to _type => TypeGraphQL.Float and the field type to number | null.

Is there a simpler way to do this?

**Comment by MichalLytek** [comment]
You can try to parse and transform them in runtime but there's no mechanisms for changing the TS and GraphQL types for generated classes.

**Reply by omar-dulaimi** [reply]
@MichalLytek Is it possible to have this as a feature request?
Since I have seen some people who have faced this issue and Prisma can't really fix it anytime soon.

**Reply by MichalLytek** [reply]
@omar-dulaimi Of course you can request a feature. However I doubt this will be resolved anytime soon.
